### PR TITLE
UISINVCOMP-34: Get all locations in `useCommonData` hook when user is in consortium mode

### PR DESCRIPTION
**Pull request by OleksandrHladchenko1**
## Purpose

* User from central tenant should see the location of member tenant holding and item.

## Approach

* Create new array of unique locations (`locationsFromAllTenants` and `locations`).

## Links
[UISINVCOMP-34](https://folio-org.atlassian.net/browse/UISINVCOMP-34)

### Before
https://github.com/user-attachments/assets/5463fb9b-1581-4701-a2d7-4e7af12342b0

### After
https://github.com/user-attachments/assets/68ec7de1-bab1-4e87-a2f2-36441f602384

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Change history for stripes-inventory-components
 
+## 1.0.1 IN PROGRESS
+
+- [UISINVCOMP-34](https://folio-org.atlassian.net/browse/UISINVCOMP-34) Get all locations in useCommonData hook when user is in consortium mode.
```
Comment (BogdanDenis) [inline CHANGELOG.md:5]: `1.0.0` hasn't been released so we can include this fix there. No need to add `1.0.1` for now

File: lib/hooks/useCommonData/useCommonData.js [hunk 2]
```diff
@@ -51,7 +52,7 @@ const useCommonData = (tenantId) => {
     const loadedData = {};
 
     if (isConsortiaEnv(stripes)) {
-      loadedData.locations = locationsFromAllTenants;
+      loadedData.locations = uniqBy([...locationsFromAllTenants, ...locations], 'id');
```
Comment (BogdanDenis) [inline lib/hooks/useCommonData/useCommonData.js:55]: Won't the locations from current tenant be included in `locationsFromAllTenants?` Why do we need to add them separately?

Comment (OleksandrHladchenko1) [reply lib/hooks/useCommonData/useCommonData.js:55]: For some reason in `locationsFromAllTenants` there are not full list of locations. That's why I decided to merge `locationsFromAllTenants` and `locations`. `locations` variable contains missing locations.

Comment (mariia-aloshyna) [reply lib/hooks/useCommonData/useCommonData.js:55]: maybe there is some issue on the BE side as to why `locationsFromAllTenants` doesn't contain all locations, or in `useLocationsFromAllTenantsQuery` query. Let's investigate it more, it looks like an unnecessary additional operation or quick workaround.

Comment (OleksandrHladchenko1) [reply lib/hooks/useCommonData/useCommonData.js:55]: @mariia-aloshyna @BogdanDenis All tenant's locations become available after reindexing the locations. Closing this PR.

**Pull request closed without merging**